src/d3a/d3a_core/sim_results/endpoint_buffer.py

```diff
@@ -153,32 +128,21 @@ def _update_cumulative_grid_trades(self, area):
     def update_stats(self, area, simulation_status, eta):
         self.status = simulation_status
         self.eta = eta
-        self._update_unmatched_loads(area)
-        # Should always precede tree-summary update
-        self.price_energy_day.update(area)
-        self.cumulative_loads = {
-            "price-currency": "Euros",
-            "load-unit": "kWh",
-            "cumulative-load-price": export_cumulative_loads(area)
-        }
+        self.cumulative_loads = export_cumulative_loads(area)
 
         self._update_cumulative_grid_trades(area)
 
         self.market_bills.update(area)
         self.balancing_bills.update(area)
 
-        self._update_tree_summary(area)
         self.trade_details = generate_inter_area_trade_details(area, "past_markets")
 
-        self.device_statistics.update(area)
-
         self.file_export_endpoints(area)
-        self.energy_trade_profile = self.file_export_endpoints.traded_energy_profile
-        self.energy_trade_profile_redis = self._round_energy_trade_profile(
-            self.file_export_endpoints.traded_energy_profile_redis)
-
-        self.last_energy_trade_profile = self._round_energy_trade_profile(
-            self.file_export_endpoints.traded_energy_current)
+        self.market_unmatched_loads.update_unmatched_loads(area)
+        self.device_statistics.update(area)
+        # Should always precede tree-summary update
+        self.price_energy_day.update(area)
+        self._update_tree_summary(area)
```

Comment:
I suggest to write a small wrapper function then. I fear that this comment will not do the trick in the long run.

Reply:
Agreed, abstracting it away to a small method.